src/utxocommit.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTXOCOMMIT_H
+#define BITCOIN_UTXOCOMMIT_H
+
+#include "hash.h"
+#include "secp256k1/include/secp256k1_multiset.h"
+#include "streams.h"
+
+#include <vector>
+
+class Coin;
+class COutPoint;
+class CCoinsViewCursor;
+
+/**
+ * A Utxo Commitment
+ *
+ * This is maintained as 96-byte multiset value that uniquely defines a UTXO set
+ *
+ * It wraps the secp256k1 multiset
+ *
+ * Note that a CUtxoCommit allows "negative sets". That is
+ *
+ * CUtxoCommit set; // set is an empty set
+ * set.Remove(X);   // set is empty set "minus" X
+ * set.Add(X);      // set is an empty set
+ *
+ * This means a CUtxoCommit can both represent the total UTXO set, or a delta to
+ * the UTXO set
+*/
+class CUtxoCommit {
+private:
+    secp256k1_multiset multiset;
+
+public:
+    // Constructs empty CUtxoCommit
+    CUtxoCommit();
+
+    // Adds a TXO to a the commitment
+    void Add(const COutPoint &out, const Coin &element);
```

Comment:
Coin contains the COutPoint, and serializes it.  Why does the COutPoint need to be serialized twice?